### PR TITLE
Add Roman Numeral in Typescript format

### LINKED ISSUE #3391: Add Roman Numeral in Typescript
To request a new code snippet, please fill out the following:

Project name: Roman Numeral
Project link: https://sampleprograms.io/projects/roman-numeral
Language: Typescript

> The project link must be a link to a project on the [Sample Programs Website project list][sample-programs-project-list].
> If you would like to add a new project to the Sample Programs Website,
> please first make a pull request to the [Sample Programs Website][sample-programs-website].

[sample-programs-website]: https://github.com/TheRenegadeCoder/sample-programs-website
[sample-programs-project-list]: https://sampleprograms.io/projects/

### PR DESCRIPTION
**Pull request by nidble**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][renegade-coder]! 
For simplicity, please make sure that your pull request includes one and only one contribution.

Please fill _one_ of the sections below as applicable.
Please also add any other relevant information to the Notes section at the bottom.
You may delete or just ignore any other sections.
For more information please refer to our [contributing documentation][contributing]

## I Am Adding a New Code Snippet in an Existing Language

- [x] I fixed #3391
- [x] I named the pull request using `Add {PROJECT} in {LANGUAGE}` format

[renegade-coder]: https://therenegadecoder.com/
[contributing-plagiarism]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[contributing-new-project]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#requirements-for-a-new-project
[contributing-new-language]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#requirements-for-a-new-language
[contributing-readme]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#create-readmes
[contributing-tests-in-detail]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#tests-in-detail
[contributing]: ../CONTRIBUTING.md
[sample-programs-project-list]: https://sampleprograms.io/projects/
[contributing-modifications]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#modifying-existing-code-snippets

### CODE REVIEW

File: archive/t/typescript/roman-numeral.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+// Roman Numeral Conversion
+
+declare const process: { argv: string[] };
```
Comment (rzuckerm) [inline archive/t/typescript/roman-numeral.ts:3]: I think this needs to be removed. I took a look at other Typescript samples, and this is not present. I think it is causing [this error](https://github.com/TheRenegadeCoder/sample-programs/actions/runs/6393475716/job/17359498270?pr=3407#step:6:7698)

**Review by rzuckerm — CHANGES_REQUESTED** [review]

File: archive/t/typescript/roman-numeral.ts (hunk 1 above)
Comment (nidble) [reply archive/t/typescript/roman-numeral.ts:3]: stripped away!

File: archive/t/typescript/roman-numeral.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  // Split the string
+  const romans = romanNumeral.split("");
+  // Get valid characters
+  const validValues = Object.keys(romanNumeralValues)
+
+  // Check for invalid characters
+  for (const roman of romans) {
+    if (roman === undefined || validValues.indexOf(roman) === -1) {
+      return "Error: " + roman + " is not a valid string of roman numerals";
```
Comment (nidble) [inline archive/t/typescript/roman-numeral.ts:33]: I tried to improve the Error's message by including the invalid string. But this is also the cause of this [error](https://github.com/TheRenegadeCoder/sample-programs/actions/runs/6397954278/job/17367161399?pr=3407#step:6:7636). 
Please, let me know how you want to proceed, do you prefer this or the more general `"Error: is not a valid string of roman numerals"`?

Comment (rzuckerm) [inline archive/t/typescript/roman-numeral.ts:33]: It looks like there's still one more [failure](https://github.com/TheRenegadeCoder/sample-programs/actions/runs/6397954278/job/17367161399?pr=3407#step:6:7643).

File: archive/t/typescript/roman-numeral.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+
+// Process arguments from CLI
+const args = process.argv.slice(2);
+
+if (args.length < 1) {
+  console.log(romanNumeralConversion());
+} else {
+  args.forEach(arg => console.log(romanNumeralConversion(arg)));
+}
```
Comment (rzuckerm) [inline archive/t/typescript/roman-numeral.ts:74]: I didn't notice this before. This same only has to process a single argument. See [this article](https://sampleprograms.io/projects/roman-numeral/) for details.

**Review by rzuckerm — CHANGES_REQUESTED** [review]

File: archive/t/typescript/roman-numeral.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  // Get valid characters
+  const validValues = Object.keys(romanNumeralValues)
+
+  // Check for invalid characters
+  for (let i = 0; i < str.length; i++) {
+    const char = str.charAt(i)
+    if (char === undefined || validValues.indexOf(char) === -1) {
+      return "Error: is not a valid string of roman numerals";
```
Comment (rzuckerm) [inline archive/t/typescript/roman-numeral.ts:32]:
```suggestion
      return "Error: invalid string of roman numerals";
```

**Review by rzuckerm — APPROVED** [review]
LGTM. I fixed the error message.

**Pull request merged**